src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -840,7 +840,26 @@ public void execute() throws GitException, InterruptedException {
 
                 args.add(fastForwardMode);
                 args.add(rev.name());
-                launchCommand(args);
+
+                /* See JENKINS-45228 */
+                /* Git merge requires authentication in LFS merges, plugin does not authenticate the git merge command */
+                String defaultRemote = null;
+                try {
+                    defaultRemote = getDefaultRemote();
+                } catch (GitException e) {
+                    /* Nothing to do, just keeping defaultRemote = null */
+                }
+
+                if (defaultRemote != null && !defaultRemote.isEmpty()) {
+                    String repoUrl = getRemoteUrl(defaultRemote);
+                    StandardCredentials cred = credentials.get(repoUrl);
+                    if (cred == null) cred = defaultCredentials;
+                    launchCommandWithCredentials(args, workspace, cred, repoUrl);
```

Comment:
@MarkEWaite Providing credentials to the git command comes with a price. Is it acceptable that credentials are provided here where in most of the cases (merges for repositories without LFS) they are not required?

Reply:
@darxriggs I think it is worth the price to provide credentials for the merge operation.  The execution time increase is likely not detectable since it is a little bit of Java code to configure a few environment variables, a small credential file written to the disc, and then the actual disc intensive merge operation is performed.

Does that seem reasonable to you?

Reply:
I had all the users in mind that are also affected by this change but don't require it. They potentially might be hit by something we forgot.

It's 3-4 temporary files created and deleted based on if its a ssh or username/password credential.

I think adding a detection if LFS is used inside the repository is overkill as it would also mean to do it recursively in submodules.

If creating and removing these temporary files is very stable and side effects are understood than it's ok for me. I just don't know the history and issues over the past years that resulted in all the code for providing the credentials.

Reply:
Creating and removing the temporary files has been stable and side effects are well understood.  Issues have been reported in the past that the temporary files had the wrong permissions (fixed) and that the plugin would benefit by switching to use a different method with newer CLI git versions (not yet implemented).  Those cases won't be altered by this addition.